indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -1220,6 +1220,14 @@ public void remove(final Task task)
     try {
       try {
         log.info("Removing task[%s] from activeTasks", task.getId());
+        if (findLocksForTask(task).stream().anyMatch(lock -> lock.getType() == TaskLockType.REPLACE)) {
```

Comment:
Should we use `findReplaceLocksForTask` method instead? But it would return only non-revoked locks.